mongo/integration/unified_runner_events_helper_test.go

```diff
@@ -24,57 +24,89 @@ import (
 // pool events.
 
 var (
-	eventTypesMap = map[string]string{
+	poolEventTypesMap = map[string]string{
 		"PoolClearedEvent": event.PoolCleared,
 	}
 	defaultCallbackTimeout = 10 * time.Second
+	sdamEventCount         = make(map[string]int)
+	sdamEventCountLock     = sync.Mutex{}
 )
 
 // unifiedRunnerEventMonitor monitors connection pool-related events.
 type unifiedRunnerEventMonitor struct {
-	sync.Mutex
-	poolEventCount map[string]int
+	poolEventCountLock sync.Mutex
+	poolEventCount     map[string]int
+	sdamMonitor        *event.ServerMonitor
 }
 
 func newUnifiedRunnerEventMonitor() *unifiedRunnerEventMonitor {
+	// reset SDAM event count for each new unifiedRunnerEventMonitor.
+	sdamEventCountLock.Lock()
+	sdamEventCount = make(map[string]int)
+	sdamEventCountLock.Unlock()
+
 	return &unifiedRunnerEventMonitor{
 		poolEventCount: make(map[string]int),
+		sdamMonitor: &event.ServerMonitor{
+			ServerDescriptionChanged: handleServerDescriptionChanged,
+		},
+	}
+}
+
+// handleServerDescriptionChanged can be used as the event handler for ServerDescriptionChanged
+// on a ServerMonitor.
+func handleServerDescriptionChanged(e *event.ServerDescriptionChangedEvent) {
+	sdamEventCountLock.Lock()
+	defer sdamEventCountLock.Unlock()
+
+	// Spec tests only ever handle ServerMarkedUnknown ServerDescriptionChangedEvents for the time being.
+	if e.NewDescription.Kind == description.Unknown {
+		sdamEventCount["ServerMarkedUnknownEvent"]++
 	}
 }
 
-// handlePoolEvent can be used as the event handler for an connection pool monitor.
+// handlePoolEvent can be used as the event handler for a connection pool monitor.
 func (u *unifiedRunnerEventMonitor) handlePoolEvent(evt *event.PoolEvent) {
-	u.Lock()
-	defer u.Unlock()
+	u.poolEventCountLock.Lock()
+	defer u.poolEventCountLock.Unlock()
 
 	u.poolEventCount[evt.Type]++
 }
 
-// getCount returns the number of events of the given type, or 0 if no events were recorded.
-func (u *unifiedRunnerEventMonitor) getCount(eventType string) int {
-	u.Lock()
-	defer u.Unlock()
+// getPoolEventCount returns the number of pool events of the given type, or 0 if no events were recorded.
+func (u *unifiedRunnerEventMonitor) getPoolEventCount(eventType string) int {
+	u.poolEventCountLock.Lock()
+	defer u.poolEventCountLock.Unlock()
 
-	mappedType := eventTypesMap[eventType]
+	mappedType := poolEventTypesMap[eventType]
 	return u.poolEventCount[mappedType]
 }
 
+// getSDAMEventCount returns the number of SDAM events of the given type, or 0 if no events were recorded.
```

Comment:
Suggest renaming this to `getServerMarkedUnknownEventCount` to clarify this only applies to those SDAM events.

Reply:
Changed to `getServerMarkedUnknownCount` and replaced `sdamEventCount` with a single `int` `serverMarkedUnknownCount`. We can switch back to a map if new spec tests introduce new SDAM event count assertions.